shim/batch.go

```diff
@@ -0,0 +1,80 @@
+// Copyright the Hyperledger Fabric contributors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package shim
+
+import (
+	"github.com/hyperledger/fabric-protos-go-apiv2/peer"
+)
+
+type batchKey struct {
+	Collection string
+	Key        string
+	Type       peer.WriteRecord_Type
```

Comment:
It won't work. You have confused the type of operation and the type of key here. In this case it should be by key type. I can do several put and delete on the same key in one transaction

Reply:
Yes, you're absolutely correct. My bad. I'll fix it

Reply:
I will ask you to delete my constants then (prefixStateDataWriteBatch prefixMetaDataWriteBatch). They don't seem to be used anywhere else in your variant